dev/flake.nix

```diff
@@ -32,6 +33,31 @@
             nixpkgs-fmt.enable = true;
           };
         };
+
+        # TODO: Move this under `CIApps` once `omnix` supports running them automatically
+        apps.cz-check = rec {
+          meta.description = program.meta.description;
+          program = pkgs.writeShellApplication {
+            name = "cz-check";
+            runtimeInputs = with pkgs; [ config.pre-commit.settings.tools.commitizen git gnugrep coreutils ];
+            meta.description = "Verify commit messages from the default branch to HEAD follow conventional commit format";
+            text = ''
+              default_branch=$(git remote show origin | grep 'HEAD branch' | cut -d' ' -f5)
+
+              # Get the latest commit on the default branch
+              # rev range using the `default_branch` branch name (e.g. main..HEAD), doesn't work in Github Actions
+              latest_default_commit=$(git rev-parse origin/"$default_branch")
+
+              current_commit=$(git rev-parse HEAD)
+
+              if [ "$latest_default_commit" = "$current_commit" ]; then
+                  echo "No commits to check between $default_branch and HEAD."
+              else
+                  cz check --rev-range "$latest_default_commit"..HEAD
+              fi
```

Comment:
This implementation isn’t generic enough, i.e, it always assumes that the current branch is going to be merged to `main`, which is okay for now. We can re-think this later if we have a complicated branch tree.

Reply:
`$GITHUB_BASE_REF`? 

https://stackoverflow.com/a/65656616/55246